src/Core/Types/Types/Scalars/UploadType.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using HotChocolate.Language;
+using HotChocolate.Properties;
+
+namespace HotChocolate.Types
+{
+    /// <summary>
+    /// The `Upload` scalar type represents a file upload.
+    ///
+    /// https://github.com/jaydenseric/graphql-multipart-request-spec
+    /// </summary>
+    [SpecScalar]
+    public sealed class UploadType : ScalarType
```

Comment:
If we introduce a new scalar then this scalar must have literals. Basically, the scalar must be bound to one of the literals defined in the spec. I am not yet convinced that we really need this one.

Reply:
Interesting. I will double check with Apollo and GrahQL-Java.

Reply:
```javascript
export const GraphQLUpload = new GraphQLScalarType({
  name: 'Upload',
  description: 'The `Upload` scalar type represents a file upload.',
  parseValue: value => value,
  parseLiteral() {
    throw new Error('‘Upload’ scalar literal unsupported.')
  },
  serialize() {
    throw new Error('‘Upload’ scalar serialization unsupported.')
  }
})
```

They are also not fully implementing the scalar.

I really do not like that, since it breaks how scalars work. We could implement the literals and say that the literal is a base64 string ... so basically it is using a string literal.

Same goes for serialize, this could also be fully implemented.

I will have to think about how we should tackle this. The upload spec feels kind of hacky.